docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1105 @@
+
+## Proposed Solution
+
+This solution uses the existing BGP feature to advertise each node's allocated
+pod subnet. BGP routers then propagate these routes throughout the provider
+network, enabling direct pod-to-pod traffic without an overlay. A single
+cluster can simultaneously have networks in both overlay and no-overlay modes.
+
+When no-overlay mode is enabled for a network:
+
+* For north-south traffic, we will follow the existing implementation of the BGP
+  feature; the traffic from pods will no longer be SNATed on egress. In the BGP
+  feature, SNAT is disabled for all pod egress traffic that leaves the node,
+  regardless of whether the destination route is BGP learnt or not. For users
+  who only want to have no-overlay mode but do not want to expose pod IPs
+  externally, this behavior is not ideal. To address this, users will be able
+  to enable SNAT for pod-to-external traffic. This will be configured via a
+  new `outboundSNAT` field in the `ClusterUserDefinedNetwork` CRD for CUDNs,
+  and the `--default-network-nat-outbound` flag for the default network.
+
+* For east-west traffic:
+  * Intra-node traffic remains unchanged, handled by the local OVN bridge.
+  * Inter-node traffic will follow the same path as north-south traffic, routed
+    via the underlay. Pod-to-pod, Pod-to-clusterIP traffic will be forwarded by
+    the provider network without encapsulation or NAT. Pod-to-node egress
+    traffic will be SNATed to the node IP at the source node to allow a pod to
+    access nodePort services in a different UDN.
+
+For the default network, we will add new flags `--default-network-overlay` and
+`--default-network-nat-outbound` to the `ovnkube` binary.
+
+* The `--default-network-overlay` flag accepts either `none` or `geneve`,
+  defaulting to `geneve`. Setting it to `none` configures the default network to
+  operate in no-overlay mode. More overlay encapsulation methods can be
+  supported in the future. If `--default-network-overlay=none` is set but no
+  `RouteAdvertisements` CR is configured for the default network, `ovnkube` will
+  exit with an error.
+* The `--default-network-nat-outbound` flag can only be set when
+  `--default-network-overlay=none`. It accepts `Enable` or `Disable` and
+  defaults to `Disable`. When set to `Enable`, it configures the default network
+  to SNAT pod-to-external traffic to the node IP. Pod-to-remote-pod traffic is not
+  SNATed, regardless of this setting.
+
+For CUDNs, no-overlay mode is configured via their respective CRs. Due to a BGP
+limitation, CUDNs with overlapping pod subnets cannot be advertised to the same
+VRF. To enable no-overlay mode for these networks, a VRF-lite topology is required.
+
+### API Details
+
+We introduce new fields to the `spec.network` section of the
+ClusterUserDefinedNetwork (CUDN) CRD to control network encapsulation:
+
+* **`overlay`**: Specifies the encapsulation method.
+  * Supported values: `Geneve` (default) or `None` (no-overlay).
```

Comment:
@pliurh : thinking out loud.. on calling it `no-overlay` -> today we assume its no-overlay with BGP since scope of RA is just for BGP - although nothing prevents us from adding OSPF tomorrow and the name `RouteAvertisement` is generic enough that I don't know if we might support more protocols. Or maybe it would be new API `LinkStateAdvertisement`
In that case, if we keep no-overlay as the name, then inside noOverlayOptions I hope we can have enough granularity for each type of no-overlay? OR do we want to be more specific around this?